src/common/task_runner.cpp

```diff
@@ -75,9 +75,9 @@ void TaskRunner::Post(const Task<void> aTask)
     Post(Milliseconds::zero(), std::move(aTask));
 }
 
-void TaskRunner::Post(Milliseconds aDelay, const Task<void> aTask)
+TaskRunner::TaskId TaskRunner::Post(Milliseconds aDelay, const Task<void> aTask)
 {
-    PushTask(aDelay, std::move(aTask));
+    return PushTask(aDelay, std::move(aTask));
```

Comment:
Not related to this PR, but it cannot trigger the move constructor because `aTask` is a `const Task<void> &&` after `std::move` instead of a **non-const** rvalue reference.

Reply:
Will do later.